src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs

```diff
@@ -502,57 +502,152 @@ private string JoinPackageNames(IEnumerable<Package> pkgs)
+                if (duplicatePackage.BuiltInPackage)
                 {
-                    var message = string.Format(Resources.MessageSamePackageSameVersInBuiltinPackages, packageToDownload);
+                    if (package.version == duplicatePackage.VersionName)
+                    {
+                        var message = string.Format(Resources.MessageSamePackageSameVersInBuiltinPackages, packageToDownload);
+
+                        MessageBoxService.Show(message, Resources.CannotDownloadPackageMessageBoxTitle,
+                            MessageBoxButton.OK, MessageBoxImage.Information);
+                    }
+                    else
+                    {
+                        var message = string.Format(Resources.MessageSamePackageDiffVersInBuiltinPackages,
+                            packageToDownload, JoinPackageNames(new[] { duplicatePackage }),
+                            DynamoViewModel.BrandingResourceProvider.ProductName);
+
+                        MessageBoxService.Show(message, Resources.CannotDownloadPackageMessageBoxTitle,
+                            MessageBoxButton.OK, MessageBoxImage.Exclamation);
+                    }
+                    return false;
+                }
 
+                if (package.version == duplicatePackage.VersionName)
+                {
+                    var message = string.Format(Resources.MessageSamePackageSameVersInLocalPackages, packageToDownload);
                     MessageBoxService.Show(message, Resources.CannotDownloadPackageMessageBoxTitle,
-                        MessageBoxButton.OK, MessageBoxImage.Information);
+                    MessageBoxButton.OK, MessageBoxImage.Information);
                 }
                 else
                 {
-                    var message = string.Format(Resources.MessageSamePackageDiffVersInBuiltinPackages,
-                        packageToDownload, JoinPackageNames(new[] { duplicatePackage }),
-                        DynamoViewModel.BrandingResourceProvider.ProductName);
+                    var dupPkg = JoinPackageNames(new[] { duplicatePackage });
+                    var message = string.Format(Resources.MessageSamePackageDiffVersInLocalPackages, packageToDownload,
+                        dupPkg, DynamoViewModel.BrandingResourceProvider.ProductName);
 
-                    MessageBoxService.Show(message, Resources.CannotDownloadPackageMessageBoxTitle,
-                        MessageBoxButton.OK, MessageBoxImage.Exclamation);
+                    var dialogResult = MessageBoxService.Show(message, Resources.CannotDownloadPackageMessageBoxTitle,
+                        MessageBoxButton.OKCancel, MessageBoxImage.Exclamation);
+
+                    if (dialogResult == MessageBoxResult.OK)
+                    {
+                        // mark for uninstallation
+                        duplicatePackage.MarkForUninstall(DynamoViewModel.Model.PreferenceSettings);
```

Comment:
should we be returning with true here ?

Reply:
I don't think so as we will not be going ahead with the download.

Reply:
What when `duplicatePackage` has no assemblies loaded. I think this may be a regression ...

Reply:
Again, here it could complicate things as well as we could end up with multiple nodes of the same name from different package versions. However, if this is confirmed to be a regression, then I agree, let's not break it - check if it's a non-assembly package, then install it (return true).